lib/archethic/bootstrap/network_constraints.ex

```diff
@@ -0,0 +1,89 @@
+defmodule Archethic.Bootstrap.NetworkConstraints do
+  @moduledoc false
+
+  alias Archethic.{
+    Reward,
+    SharedSecrets,
+    OracleChain
+  }
+
+  require Logger
+
+  @spec persist_genesis_address() :: :ok
+  def persist_genesis_address() do
+    res = process_gen_addr()
+
+    Logger.debug(
+      "Gen Addr: reward: #{res.reward}, origin: #{res.origin}, nss: #{res.node_shared_secrets}"
+    )
+
+    if Enum.all?(res, fn {_k, v} -> v == :ok end) do
+      Logger.info("Genesis Address Loading: Successful")
+    else
+      Logger.info("Genesis Address Loading: Failed, Resheduled to Next-Self-Repair")
+    end
+
+    :ok
+  end
+
+  @spec process_gen_addr() :: map()
+  def process_gen_addr() do
```

Comment:
I think you could either make it private or embed it into `persiste_genesis_address`